xmtp_api_http/src/lib.rs

```diff
@@ -57,26 +66,16 @@ pub struct XmtpHttpApiClient {
     host_url: String,
     app_version: String,
     libxmtp_version: String,
+    limiter: Arc<Limiter>,
     stats: ApiStats,
     identity_stats: IdentityStats,
 }
 
 impl XmtpHttpApiClient {
-    pub fn new(host_url: String, app_version: String) -> Result<Self, HttpClientError> {
-        let client = reqwest_builder();
-        let libxmtp_version = env!("CARGO_PKG_VERSION").to_string();
-        let mut headers = header::HeaderMap::new();
-        headers.insert("x-libxmtp-version", libxmtp_version.parse()?);
-        let client = client.default_headers(headers).build()?;
-
-        Ok(XmtpHttpApiClient {
-            http_client: client,
-            host_url,
-            app_version,
-            libxmtp_version,
-            stats: ApiStats::default(),
-            identity_stats: IdentityStats::default(),
-        })
+    /// Wait for any rate limit
+    async fn wait_for_ready(&self) {
+        let jitter = Jitter::up_to(Duration::from_secs(5));
```

Comment:
What are the impacts of the jitter here?

Reply:
from the docs:

```
When polled, the returned future either resolves immediately (in the case where the rate limiter allows it), or else triggers an asynchronous delay, after which the rate limiter is polled again. This means that the future might resolve at some later time (depending on what other measurements are made on the rate limiter).

This method allows for a randomized additional delay between polls of the rate limiter, which can help reduce the likelihood of thundering herd effects if multiple tasks try to wait on the same rate limiter.
```

so it seems like another randomized value of up to 5 seconds is added to the delay, which is one minute

Reply:
I guess what I'm wondering is if it can add a delay to calls when you aren't already rate limited

Reply:
ah good catch, i didn't know but checking the implementation [here](https://docs.rs/governor/latest/src/governor/state/direct/future.rs.html#51) seems like the delay is only added if the limiter check fails/we are being rate limited